{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSServerError"}],"languages":["swift"],"platforms":["macOS"]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/sotocore\/awsservererror"],"traits":[{"interfaceLanguage":"swift"}]}],"relationshipsSections":[{"identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","doc:\/\/codes.soto.soto-core\/SQ","doc:\/\/codes.soto.soto-core\/s5ErrorP","doc:\/\/codes.soto.soto-core\/s8SendableP"],"kind":"relationships","title":"Conforms To","type":"conformsTo"}],"identifier":{"url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","interfaceLanguage":"swift"},"topicSections":[{"title":"Initializers","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/init(errorCode:context:)"]},{"title":"Instance Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/context","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/errorCode"]},{"title":"Type Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/internalFailure","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/serviceUnavailable"]},{"title":"Default Implementations","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/AWSErrorType-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/CustomStringConvertible-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Equatable-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Error-Implementations"],"generated":true}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSServerError"}],"title":"AWSServerError","roleHeading":"Structure","role":"symbol","symbolKind":"struct","externalID":"s:8SotoCore14AWSServerErrorV","modules":[{"name":"SotoCore"}],"navigatorTitle":[{"kind":"identifier","text":"AWSServerError"}]},"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore"]]},"seeAlsoSections":[{"title":"Errors","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError"],"generated":true}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore":{"role":"collection","title":"SotoCore","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","kind":"symbol","type":"topic","url":"\/documentation\/sotocore"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorContext":{"role":"symbol","title":"AWSErrorContext","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorContext"}],"abstract":[{"type":"text","text":"Additional information about error"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSErrorContext"}],"url":"\/documentation\/sotocore\/awserrorcontext"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType":{"role":"symbol","title":"AWSErrorType","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorType"}],"abstract":[{"type":"text","text":"Standard Error type returned by Soto."}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSErrorType"}],"url":"\/documentation\/sotocore\/awserrortype"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/AWSErrorType-Implementations":{"role":"collectionGroup","title":"AWSErrorType Implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/AWSErrorType-Implementations","kind":"article","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/awserrortype-implementations"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSRawError":{"role":"symbol","title":"AWSRawError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSRawError"}],"abstract":[{"type":"text","text":"Raw unprocessed error."}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSRawError"}],"url":"\/documentation\/sotocore\/awsrawerror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError":{"role":"symbol","title":"AWSResponseError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSResponseError"}],"abstract":[{"type":"text","text":"Response Error type returned by Soto if the error code is unrecognised"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSResponseError"}],"url":"\/documentation\/sotocore\/awsresponseerror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/errorCode":{"role":"symbol","title":"errorCode","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"errorCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"abstract":[{"type":"text","text":"return error code string"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/errorCode","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/errorcode"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSClientError":{"role":"symbol","title":"AWSClientError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSClientError"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSClientError"}],"url":"\/documentation\/sotocore\/awsclienterror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError":{"role":"symbol","title":"AWSServerError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSServerError"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSServerError"}],"url":"\/documentation\/sotocore\/awsservererror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/internalFailure":{"role":"symbol","title":"internalFailure","fragments":[{"kind":"keyword","text":"static"},{"kind":"text","text":" "},{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"internalFailure"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"AWSServerError","preciseIdentifier":"s:8SotoCore14AWSServerErrorV"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/internalFailure","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/internalfailure"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/serviceUnavailable":{"role":"symbol","title":"serviceUnavailable","fragments":[{"kind":"keyword","text":"static"},{"kind":"text","text":" "},{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"serviceUnavailable"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"AWSServerError","preciseIdentifier":"s:8SotoCore14AWSServerErrorV"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/serviceUnavailable","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/serviceunavailable"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/Equatable-Implementations":{"role":"collectionGroup","title":"Equatable Implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Equatable-Implementations","kind":"article","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/equatable-implementations"},"doc://codes.soto.soto-core/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/codes.soto.soto-core\/s8SendableP"},"doc://codes.soto.soto-core/s23CustomStringConvertibleP":{"type":"unresolvable","title":"Swift.CustomStringConvertible","identifier":"doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/CustomStringConvertible-Implementations":{"role":"collectionGroup","title":"CustomStringConvertible Implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/CustomStringConvertible-Implementations","kind":"article","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/customstringconvertible-implementations"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/init(errorCode:context:)":{"role":"symbol","title":"init(errorCode:context:)","fragments":[{"kind":"identifier","text":"init"},{"kind":"text","text":"?("},{"kind":"externalParam","text":"errorCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":", "},{"kind":"externalParam","text":"context"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"AWSErrorContext","preciseIdentifier":"s:8SotoCore15AWSErrorContextV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"initialize AWSServerError"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/init(errorCode:context:)","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/init(errorcode:context:)"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/context":{"role":"symbol","title":"context","fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"context"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"AWSErrorContext","preciseIdentifier":"s:8SotoCore15AWSErrorContextV"},{"kind":"text","text":"?"}],"abstract":[{"type":"text","text":"additional context information related to the error"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/context","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/context"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/Error-Implementations":{"role":"collectionGroup","title":"Error Implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Error-Implementations","kind":"article","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/error-implementations"},"doc://codes.soto.soto-core/SQ":{"type":"unresolvable","title":"Swift.Equatable","identifier":"doc:\/\/codes.soto.soto-core\/SQ"},"doc://codes.soto.soto-core/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/codes.soto.soto-core\/s5ErrorP"}}}